pkg/network/setup/netconf_test.go

```diff
@@ -49,39 +49,58 @@ var _ = Describe("netconf", func() {
 	})
 
 	It("runs setup successfully", func() {
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupDummyNoop)).To(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeTrue())
-	})
-
-	It("runs teardown successfully", func() {
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupDummyNoop)).To(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeTrue())
-		Expect(netConf.Teardown(vmi)).To(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeFalse())
-	})
-
-	It("skips secondary setup runs", func() {
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupDummyNoop)).To(Succeed())
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupFail)).To(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeTrue())
+		Expect(netConf.Setup(vmi, vmi.Spec.Networks, launcherPid, netPreSetupDummyNoop)).To(Succeed())
 	})
 
 	It("fails the pre-setup run", func() {
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupFail)).NotTo(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeFalse())
+		Expect(netConf.Setup(vmi, vmi.Spec.Networks, launcherPid, netPreSetupFail)).NotTo(Succeed())
 	})
 
 	It("fails the setup run", func() {
 		netConf := netsetup.NewNetConfWithCustomFactory(nsFailureFactory, &tempCacheCreator{})
-		Expect(netConf.Setup(vmi, launcherPid, netPreSetupDummyNoop)).NotTo(Succeed())
-		Expect(netConf.SetupCompleted(vmi)).To(BeFalse())
+		Expect(netConf.Setup(vmi, vmi.Spec.Networks, launcherPid, netPreSetupDummyNoop)).NotTo(Succeed())
 	})
 
 	It("fails the teardown run", func() {
 		netConf := netsetup.NewNetConfWithCustomFactory(nil, failingCacheCreator{})
 		Expect(netConf.Teardown(vmi)).NotTo(Succeed())
 		Expect(netConf.SetupCompleted(vmi)).To(BeFalse())
 	})
+
+	Context("with completion cache", func() {
+		It("runs setup successfully", func() {
```

Comment:
Please add the same test to the without `WithCompletionCache` tests section.

Reply:
There is one: The first test at line 51.